src/ott/solvers/outputs.py

```diff
@@ -0,0 +1,66 @@
+from typing import Optional, Tuple
+
+import jax.numpy as jnp
+from flax import struct
+
+from ott.geometry import geometry
+from ott.problems.linear import linear_problem
+
+
+@struct.dataclass
+class _BaseTransportOutput:
+  """Base class for Transport output.
+
+  Args:
+    ot_prob: OT problem.
+    errors: Holds sequence of vectors of errors of the Sinkhorn algorithm
+      at each iteration.
+    shape: Shape of the transport plan.
+    costs: Holds the sequence of regularized GW costs seen through the loop
+      of the solver.
+    converged: Boolean value wether or not the problem has converged.
+    reg_ot_cost: Value of the regularised OT loss.
+  """
+
+  # TODO(michalk8): must be called `errors`, because of `store_inner_errors`
+  # in future, enforce via class hierarchy
+  ot_prob: linear_problem.LinearProblem
+  errors: jnp.ndarray
+  shape: Tuple[int, int] = struct.field(pytree_node=False)
```

Comment:
just a quick comment. Usually shape is already stored in `ot_prob.geom.shape`, so why not define it to be a property that points to that directly? I am worried about inconsistencies that may arise.